### PR TITLE
Remove Channel::cid from constructor

### PR DESCRIPTION
**Pull request by adasiewiczr**
### 🎯 Goal

 Removed `cid` from the channel's related objects' constructors. It's now an immutable property calculated based on `type` and `id`. 

### 🛠 Implementation details
- Converted `Channel:cid` to an immutable property
- Converted `ChannelData:cid` to an immutable property
- Renamed `ChannelData:channelId` to `ChannelData::id`

### 🧪 Testing
Run both sample apps and:
- Verify channels are fetched properly
- You can send messages
- You can create a channel

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Channel.kt [hunk 1]
```diff
@@ -90,6 +88,16 @@ public data class Channel(
...
 
+    /**
+     * The channel id in the format messaging:123.
+     */
+    val cid: String
+        get() = if (id.isEmpty() || type.isEmpty()) {
+            ""
+        } else {
+            "$type:$id"
+        }
```
Comment (JcMinarro) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Channel.kt:99]: What is the problem returning always `$type:$id`? :thinking:

File: stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/model/channel/ChannelData.kt [hunk 2]
```diff
@@ -64,6 +62,16 @@ public data class ChannelData(
...
 
+    /**
+     * The channel id in the format messaging:123.
+     */
+    val cid: String
+        get() = if (id.isEmpty() || type.isEmpty()) {
+            ""
+        } else {
+            "$type:$id"
+        }
```
Comment (JcMinarro) [inline stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/model/channel/ChannelData.kt:73]: Same here

**Review by JcMinarro — APPROVED** [review]
Great change!!!! :partying_face: :tada: 
I only left a comment, could you review it?

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Channel.kt (hunk 1 above)
Comment (adasiewiczr) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Channel.kt:99]: 1. I think it's better to simply return `""` when either type or id is empty as we are not able to create valid cid
2. In Compose we have this code:
```val channel = selectedChannel ?: Channel()
AnimatedVisibility(
     visible = channel.cid.isNotEmpty(), // not empty here, cid is ":" here
     enter = fadeIn(),
     exit = fadeOut(animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2))
) {
}```
for showing the channel's options.

Comment (bychkovdmitry) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/models/Channel.kt:99]: Yep, for convenience not to deal with nullable strings we do:

```
val channel = selectedChannel ?: Channel()
AnimatedVisibility(
      visible = channel.cid.isNotEmpty(),
     enter = fadeIn(),
      exit = fadeOut(animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2))
)
``` 
And "$type:$id" will evaluate to ":" which breaks our UI.

**Pull request merged**